master/internal/experiment.go

```diff
@@ -1033,6 +1034,29 @@ func (e *internalExperiment) setPriority(priority *int, forward bool) (err error
 	if priority == nil {
 		return nil
 	}
+
+	workspaceModel, err := workspace.WorkspaceByProjectID(context.TODO(), e.ProjectID)
+	if err != nil && errors.Cause(err) != sql.ErrNoRows {
+		return err
+	}
+	wkspID := resolveWorkspaceID(workspaceModel)
+
+	// Returns an error if RM does not implement priority.
+	if smallerHigher, err := e.rm.SmallerValueIsHigherPriority(); err == nil {
+		ok, err := configpolicy.PriorityAllowed(
```

Comment:
This looks great! Same comment as [here](https://github.com/determined-ai/determined/pull/10047#discussion_r1799456338)